tiledb/sm/group/group.h

```diff
@@ -58,7 +60,23 @@ class GroupDetailsException : public StatusException {
 
 class Group {
  public:
-  Group(const URI& group_uri, StorageManager* storage_manager);
+  /**
+   * Constructs a Group object given a uri and a ContextResources reference.
+   * This is a transitional constructor in the sense that we are working
+   * on removing the dependency of the Group class on StorageManager. For
+   * now we still need to keep the storage_manager argument, but once the
+   * dependency is gone the signature will be
+   * Group(ContextResources&, const URI&).
```

Comment:
Future: We'll need a configuration argument as well. Probably should change the comment.

Will still approve as is, but worth a tiny update.